### ISSUE TITLE
[BUG] Repeater values in bolt_field_values after the repeater has been deleted causing mysql to hang

### ISSUE THREAD

**Issue by graham73may** [bolt/bolt#6276]
Details
-------

 - Relevant Bolt Version: 3.2.3
 - Install type: Composer install
 - PHP version: 5.6
 - Used web server: Apache

Reproduction
------------
​
 1. **Bug summary**: 
We've got an interesting situation, not sure if this is a bug... we can't really decide what *should* happen.

If you have a repeater with content in the DB, and you change the name. All hell breaks loose and the Bolt repeater query stalls and has taken down mysql a couple of times this morning on our test server.

Our fix was to delete the now unused data from the `bolt_field_values` table.

The symptom of crashing mysql is quite severe.

 2. **Specifics**:
```sql
SELECT content.*, 
       Group_concat(incomingrelation.id) 
       AS _incomingrelation_id, 
       Group_concat(incomingrelation.from_id) 
       AS _incomingrelation_fromid, 
       Group_concat(incomingrelation.from_contenttype) 
       AS 
       _incomingrelation_fromcontenttype, 
       Group_concat(tags.id ORDER BY tags.id ASC) 
       AS _tags_id, 
       Group_concat(tags.slug ORDER BY tags.id ASC) 
       AS _tags_slug, 
       Group_concat(tags.name ORDER BY tags.id ASC) 
       AS _tags_name, 
       Group_concat(tags.taxonomytype ORDER BY tags.id ASC) 
       AS _tags_taxonomytype, 
       Group_concat(DISTINCT Concat_ws('_', f_tabs.name, f_tabs.grouping, 
       f_tabs.id)) 
                 AS tabs, 
       Group_concat(DISTINCT Concat_ws('_', f_blocks.name, f_blocks.grouping, 
       f_blocks.id)) 
       AS blocks, 
       Group_concat(DISTINCT Concat_ws('_', f_ft_content_cards.name, 
       f_ft_content_cards.grouping, 
       f_ft_content_cards.id)) 
       AS ft_content_cards, 
       Group_concat(DISTINCT Concat_ws('_', f_ft_content_links.name, 
       f_ft_content_links.grouping, 
       f_ft_content_links.id)) 
       AS ft_content_links, 
       Group_concat(DISTINCT Concat_ws('_', 
       f_ft_content_upcoming_events_cards.name, 
       f_ft_content_upcoming_events_cards.grouping, 
       f_ft_content_upcoming_events_cards.id)) 
       AS ft_content_upcoming_events_cards, 
       Group_concat(DISTINCT Concat_ws('_', 
       f_ft_content_upcoming_events_links.name, 
       f_ft_content_upcoming_events_links.grouping, 
       f_ft_content_upcoming_events_links.id)) 
       AS ft_content_upcoming_events_links, 
       Group_concat(DISTINCT Concat_ws('_', 
       f_ft_content_recent_events_cards.name, 
       f_ft_content_recent_events_cards.grouping, 
       f_ft_content_recent_events_cards.id)) 
                                         AS ft_content_recent_events_cards, 
       Group_concat(DISTINCT Concat_ws('_', 
       f_ft_content_recent_events_links.name, 
       f_ft_content_recent_events_links.grouping, 
       f_ft_content_recent_events_links.id)) 
                                         AS ft_content_recent_events_links, 
       Group_concat(DISTINCT Concat_ws('_', 
       f_ft_content_learning_networks_cards.name, 
       f_ft_content_learning_networks_cards.grouping, 
       f_ft_content_learning_networks_cards.id)) 
       AS ft_content_learning_networks_cards, 
       Group_concat(DISTINCT Concat_ws('_', 
       f_ft_content_learning_networks_links.name, 
       f_ft_content_learning_networks_links.grouping, 
       f_ft_content_learning_networks_links.id)) 
       AS ft_content_learning_networks_links, 
       Group_concat(DISTINCT Concat_ws('_', f_ft_content_comments_cards.name, 
       f_ft_content_comments_cards.grouping, f_ft_content_comments_cards.id)) 
       AS 
       ft_content_comments_cards, 
       Group_concat(DISTINCT Concat_ws('_', f_ft_content_comments_links.name, 
       f_ft_content_comments_links.grouping, f_ft_content_comments_links.id)) 
       AS 
       ft_content_comments_links, 
       Group_concat(DISTINCT Concat_ws('_', f_ft_content_latest_news_cards.name, 
       f_ft_content_latest_news_cards.grouping, 
       f_ft_content_latest_news_cards.id)) AS 
       ft_content_latest_news_cards, 
       Group_concat(DISTINCT Concat_ws('_', f_ft_content_latest_news_links.name, 
       f_ft_content_latest_news_links.grouping, 
       f_ft_content_latest_news_links.id)) AS 
       ft_content_latest_news_links, 
       Group_concat(DISTINCT Concat_ws('_', f_page_links.name, 
       f_page_links.grouping, 
       f_page_links.id)) 
       AS page_links 
FROM   bolt_pages content 
       LEFT JOIN bolt_relations incomingrelation 
              ON content.id = incomingrelation.to_id 
                 AND incomingrelation.to_contenttype = 'pages' 
       LEFT JOIN bolt_taxonomy tags 
              ON content.id = tags.content_id 
                 AND tags.contenttype = 'pages' 
                 AND tags.taxonomytype = 'tags' 
       LEFT JOIN bolt_field_value f_tabs 
              ON f_tabs.content_id = content.id 
                 AND f_tabs.contenttype = 'pages' 
                 AND f_tabs.name = 'tabs' 
       LEFT JOIN bolt_field_value f_blocks 
              ON f_blocks.content_id = content.id 
                 AND f_blocks.contenttype = 'pages' 
                 AND f_blocks.name = 'blocks' 
       LEFT JOIN bolt_field_value f_ft_content_cards 
              ON f_ft_content_cards.content_id = content.id 
                 AND f_ft_content_cards.contenttype = 'pages' 
                 AND f_ft_content_cards.name = 'ft_content_cards' 
       LEFT JOIN bolt_field_value f_ft_content_links 
              ON f_ft_content_links.content_id = content.id 
                 AND f_ft_content_links.contenttype = 'pages' 
                 AND f_ft_content_links.name = 'ft_content_links' 
       LEFT JOIN bolt_field_value f_ft_content_upcoming_events_cards 
              ON f_ft_content_upcoming_events_cards.content_id = content.id 
                 AND f_ft_content_upcoming_events_cards.contenttype = 'pages' 
                 AND f_ft_content_upcoming_events_cards.name = 
                     'ft_content_upcoming_events_cards' 
       LEFT JOIN bolt_field_value f_ft_content_upcoming_events_links 
              ON f_ft_content_upcoming_events_links.content_id = content.id 
                 AND f_ft_content_upcoming_events_links.contenttype = 'pages' 
                 AND f_ft_content_upcoming_events_links.name = 
                     'ft_content_upcoming_events_links' 
       LEFT JOIN bolt_field_value f_ft_content_recent_events_cards 
              ON f_ft_content_recent_events_cards.content_id = content.id 
                 AND f_ft_content_recent_events_cards.contenttype = 'pages' 
                 AND f_ft_content_recent_events_cards.name = 
                     'ft_content_recent_events_cards' 
       LEFT JOIN bolt_field_value f_ft_content_recent_events_links 
              ON f_ft_content_recent_events_links.content_id = content.id 
                 AND f_ft_content_recent_events_links.contenttype = 'pages' 
                 AND f_ft_content_recent_events_links.name = 
                     'ft_content_recent_events_links' 
       LEFT JOIN bolt_field_value f_ft_content_learning_networks_cards 
              ON f_ft_content_learning_networks_cards.content_id = content.id 
                 AND f_ft_content_learning_networks_cards.contenttype = 'pages' 
                 AND f_ft_content_learning_networks_cards.name = 
                     'ft_content_learning_networks_cards' 
       LEFT JOIN bolt_field_value f_ft_content_learning_networks_links 
              ON f_ft_content_learning_networks_links.content_id = content.id 
                 AND f_ft_content_learning_networks_links.contenttype = 'pages' 
                 AND f_ft_content_learning_networks_links.name = 
                     'ft_content_learning_networks_links' 
       LEFT JOIN bolt_field_value f_ft_content_comments_cards 
              ON f_ft_content_comments_cards.content_id = content.id 
                 AND f_ft_content_comments_cards.contenttype = 'pages' 
                 AND f_ft_content_comments_cards.name = 
                     'ft_content_comments_cards' 
       LEFT JOIN bolt_field_value f_ft_content_comments_links 
              ON f_ft_content_comments_links.content_id = content.id 
                 AND f_ft_content_comments_links.contenttype = 'pages' 
                 AND f_ft_content_comments_links.name = 
                     'ft_content_comments_links' 
       LEFT JOIN bolt_field_value f_ft_content_latest_news_cards 
              ON f_ft_content_latest_news_cards.content_id = content.id 
                 AND f_ft_content_latest_news_cards.contenttype = 'pages' 
                 AND f_ft_content_latest_news_cards.name = 
                     'ft_content_latest_news_cards' 
       LEFT JOIN bolt_field_value f_ft_content_latest_news_links 
              ON f_ft_content_latest_news_links.content_id = content.id 
                 AND f_ft_content_latest_news_links.contenttype = 'pages' 
                 AND f_ft_content_latest_news_links.name = 
                     'ft_content_latest_news_links' 
       LEFT JOIN bolt_field_value f_page_links 
              ON f_page_links.content_id = content.id 
                 AND f_page_links.contenttype = 'pages' 
                 AND f_page_links.name = 'page_links' 
WHERE  content.id = '1' 
GROUP  BY content.id, 
          content.id 
```

 3. **Steps to reproduce**:
- Create a repeater in contenttypes.yml
- Add some data using the repeater
- Rename a sub field in the repeater in contenttypes.yml
- Reload the edit page

 4. **Expected result**: 
See the edit page

 5. **Actual result**: 
Page fails to load (just hangs). `mysql` service on the server starts to overload - the more times you try to view an edit page, the worse the load gets, this is multiplied by the number of repeaters on the contenttype

**Comment by graham73may** [comment]
I'm assuming this is quite an expert level issue, but please do let me (or @mrenigma) know if there's anything that can be done to help get things moving on this one.

**Comment by graham73may** [comment]
Just a note: This may be related to #6264 (6264 has been fixed by #6280, will keep an eye on if this also has a positive effect on this issue).
